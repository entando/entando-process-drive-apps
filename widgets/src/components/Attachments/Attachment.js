import React from 'react';
import PropTypes from 'prop-types';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
import FileCopyIcon from '@material-ui/icons/FileCopy';
import DeleteIcon from '@material-ui/icons/Delete';
import IconButton from '@material-ui/core/IconButton';
import Link from '@material-ui/core/Link';
import withStyles from '@material-ui/core/styles/withStyles';

import ConfirmDialog from 'components/common/ConfirmDialog';

const styles = {
  listItem: {
    border: 'solid 1px #eee',
    marginBottom: 2,
    background: 'white',
    borderRadius: 3,
  },
  truncate: {
    display: 'block',
    overflow: 'hidden',
    width: '80%',
    textOverflow: 'ellipsis',
  },
};

class Attachment extends React.Component {
  state = {
    dialogOpen: false,
  };

  handleDeleteDialog = () => {
    this.setState(state => ({ dialogOpen: !state.dialogOpen }));
  };

  render() {
    const { dialogOpen } = this.state;
    const { classes, item, onDelete, downloadLink } = this.props;

    return (
      <>
        <ListItem className={classes.listItem}>
          <ListItemIcon>
            <FileCopyIcon fontSize="small" />
          </ListItemIcon>
          <ListItemText className={classes.truncate}>
            <Link href={`${downloadLink}/${item.id}/download`} download>
              {item.name}
            </Link>
          </ListItemText>
          <ListItemSecondaryAction>
            <IconButton size="small" onClick={this.handleDeleteDialog}>
              <DeleteIcon fontSize="small" />
            </IconButton>
          </ListItemSecondaryAction>
        </ListItem>
        <ConfirmDialog
          title="Attachment exclusion"
          message={`Do you wish to delete this file permanently?\n${item.name}`}
          open={dialogOpen}
          onClose={this.handleDeleteDialog}
          onConfirm={onDelete(item)}
          maxWidth="md"
        />
      </>
    );
  }
}

Attachment.propTypes = {
  classes: PropTypes.shape({
    listItem: PropTypes.string,
    truncate: PropTypes.string,
  }),
  item: PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string,
  }),
  onDelete: PropTypes.func.isRequired,
  downloadLink: PropTypes.string.isRequired,
};

Attachment.defaultProps = {
  classes: {},
  item: {},
};

export default withStyles(styles)(Attachment);
