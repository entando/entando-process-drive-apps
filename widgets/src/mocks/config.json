{
  "payload": {
    "code": "pda_task_list",
    "config": {
      "groups": "[{\"label\":\"Administrators\",\"key\":\"Administrators\",\"checked\":true},{\"label\":\"broker\",\"key\":\"broker\",\"checked\":true},{\"label\":\"admin\",\"key\":\"admin\",\"checked\":false},{\"label\":\"rest-all\",\"key\":\"rest-all\",\"checked\":false},{\"label\":\"kiemgmt\",\"key\":\"kiemgmt\",\"checked\":true},{\"label\":\"kie-server\",\"key\":\"kie-server\",\"checked\":true},{\"label\":\"appraiser\",\"key\":\"appraiser\",\"checked\":false},{\"label\":\"manager\",\"key\":\"manager\",\"checked\":true},{\"label\":\"supplier\",\"key\":\"supplier\",\"checked\":true},{\"label\":\"IT\",\"key\":\"IT\",\"checked\":true},{\"label\":\"PM\",\"key\":\"PM\",\"checked\":true}]",
      "options": "[{\"key\":\"newPageOnClick\",\"label\":\"Open new page on table row click\",\"checked\":false},{\"key\":\"showClaim\",\"label\":\"Show Claim Button\",\"checked\":true},{\"key\":\"showComplete\",\"label\":\"Show Complete Button\",\"checked\":true}]",
      "columns": "[{\"name\":\"task-is-skipable\",\"position\":0,\"isVisible\":false},{\"name\":\"task-parent-id\",\"position\":1,\"isVisible\":false},{\"name\":\"task-created-by\",\"position\":2,\"isVisible\":true},{\"name\":\"task-input-data.application.property.saleprice\",\"position\":3,\"isVisible\":true},{\"name\":\"task-input-data.TaskName\",\"position\":4,\"isVisible\":true},{\"name\":\"initiator\",\"position\":5,\"isVisible\":true},{\"name\":\"task-created-on\",\"position\":6,\"isVisible\":true},{\"name\":\"task-priority\",\"position\":7,\"isVisible\":false},{\"name\":\"task-proc-def-id\",\"position\":8,\"isVisible\":true},{\"name\":\"task-form\",\"position\":9,\"isVisible\":true},{\"name\":\"task-skippable\",\"position\":10,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.name\",\"position\":11,\"isVisible\":true},{\"name\":\"task-actual-owner\",\"position\":12,\"isVisible\":true},{\"name\":\"task-activation-time\",\"position\":13,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.address\",\"position\":14,\"isVisible\":true},{\"name\":\"task-input-data.application.errors\",\"position\":15,\"isVisible\":true},{\"name\":\"task-name\",\"position\":16,\"isVisible\":true},{\"name\":\"task-input-data.application.property.address\",\"position\":17,\"isVisible\":true},{\"name\":\"task-subject\",\"position\":18,\"isVisible\":true},{\"name\":\"containerId\",\"position\":19,\"isVisible\":true},{\"name\":\"task-container-id\",\"position\":20,\"isVisible\":true},{\"name\":\"processDefinitionId\",\"position\":21,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.ssn\",\"position\":22,\"isVisible\":true},{\"name\":\"task-description\",\"position\":23,\"isVisible\":true},{\"name\":\"task-input-data.Skippable\",\"position\":24,\"isVisible\":true},{\"name\":\"task-workitem-id\",\"position\":25,\"isVisible\":true},{\"name\":\"task-expiration-time\",\"position\":26,\"isVisible\":true},{\"name\":\"task-input-data.application.property.age\",\"position\":27,\"isVisible\":true},{\"name\":\"task-status\",\"position\":28,\"isVisible\":true},{\"name\":\"task-input-data.application.mortgageamount\",\"position\":29,\"isVisible\":true},{\"name\":\"task-process-instance-id\",\"position\":30,\"isVisible\":true},{\"name\":\"task-input-data.ActorId\",\"position\":31,\"isVisible\":true},{\"name\":\"task-process-id\",\"position\":32,\"isVisible\":true},{\"name\":\"task-input-data.application.property.locale\",\"position\":33,\"isVisible\":true},{\"name\":\"task-input-data.application.amortization\",\"position\":34,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.creditrating\",\"position\":35,\"isVisible\":true},{\"name\":\"application\",\"position\":36,\"isVisible\":true},{\"name\":\"task-input-data.NodeName\",\"position\":37,\"isVisible\":true},{\"name\":\"task-proc-inst-id\",\"position\":38,\"isVisible\":true},{\"name\":\"task-input-data.application.downpayment\",\"position\":39,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.annualincome\",\"position\":40,\"isVisible\":true},{\"name\":\"task-id\",\"position\":41,\"isVisible\":true}]",
      "knowledgeSource": "kieStaging",
      "process": "itorders.orderhardware@itorders_1.0.0-SNAPSHOT"
    }
  },
  "metaData": {
    "status": "draft"
  },
  "errors": []
}
