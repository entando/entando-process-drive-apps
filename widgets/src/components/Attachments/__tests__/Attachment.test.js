import React from 'react';
import { render } from '@testing-library/react';

import Attachment from '../Attachment';
import 'mocks/i18nMock';

describe('<Attachment />', () => {
  it('renders snapshot correctly', () => {
    const onDelete = () => jest.fn();
    const onDownload = () => jest.fn();
    const item = {
      id: '1',
      name: 'test',
    };
    const { container } = render(
      <Attachment item={item} onDelete={onDelete} onDownload={onDownload} />
    );

    expect(container).toMatchSnapshot();
  });
});
