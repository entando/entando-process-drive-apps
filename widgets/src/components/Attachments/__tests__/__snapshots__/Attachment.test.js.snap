// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Attachment /> renders snapshot correctly 1`] = `
<div>
  <li
    class="MuiListItem-container"
  >
    <div
      class="MuiListItem-root Attachment-listItem-1 MuiListItem-gutters MuiListItem-secondaryAction"
    >
      <div
        class="MuiListItemIcon-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4l6 6v10c0 1.1-.9 2-2 2H7.99C6.89 23 6 22.1 6 21l.01-14c0-1.1.89-2 1.99-2h7zm-1 7h5.5L14 6.5V12z"
          />
        </svg>
      </div>
      <div
        class="MuiListItemText-root Attachment-truncate-2"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          >
            test
          </a>
        </span>
      </div>
    </div>
    <div
      class="MuiListItemSecondaryAction-root"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </li>
</div>
`;
